crates/router/src/db/ephemeral_key.rs

```diff
@@ -125,21 +127,57 @@ mod storage {
 impl EphemeralKeyInterface for MockDb {
     async fn create_ephemeral_key(
         &self,
-        _ek: EphemeralKeyNew,
-        _validity: i64,
+        ek: EphemeralKeyNew,
+        validity: i64,
     ) -> CustomResult<EphemeralKey, errors::StorageError> {
-        Err(errors::StorageError::KVError.into())
+        let mut ephemeral_keys = self.ephemeral_keys.lock().await;
+        let created_at = common_utils::date_time::now();
+        let expires = created_at.saturating_add(validity.hours());
```

Comment:
That method is available because of the [`NumericalDuration`](https://docs.rs/time/latest/time/ext/trait.NumericalDuration.html#) trait being in scope. Marking this conversation resolved.